prowler/providers/aws/services/ec2/ec2_service.py

```diff
@@ -223,6 +224,25 @@ def __get_snapshot_public__(self):
     def __describe_network_interfaces__(self, regional_client):
         logger.info("EC2 - Describing Network Interfaces...")
         try:
+            # Get IPs for network interfaces
+            describe_network_interfaces_paginator = regional_client.get_paginator(
+                "describe_network_interfaces"
+            )
+            for page in describe_network_interfaces_paginator.paginate():
+                for interface in page["NetworkInterfaces"]:
+                    if interface.get("Association"):
+                        self.network_interfaces.append(
+                            NetworkInterface(
+                                public_ip=interface["Association"]["PublicIp"],
+                                type=interface["InterfaceType"],
+                                private_ip=interface["PrivateIpAddress"],
+                                subnet_id=interface["SubnetId"],
+                                vpc_id=interface["VpcId"],
+                                region=regional_client.region,
+                                tags=interface.get("TagSet"),
+                            )
+                        )
+
```

Comment:
Why are we calling `describe_network_interfaces` twice? One here and another time down below.

Reply:
The other one is used to retrieve the interfaces for a specific security group.

Reply:
The other one is used to retrieve the interfaces for a specific security group.

Reply:
In that case I see two options:
1. Split this into two functions
2. Make the search conditional to save the SG network interface during the first search

Reply:
I've done option 1